### ISSUE TITLE
Q: Which one better to use?

### ISSUE THREAD

**Issue by mnbv09870987** [PerlFFI/FFI-Platypus#273]
Sorry for question (is it OK to ask them here?)

Do I understand that to define next struct it would be better if use `FFI::P::Record` instead of `FFI::C`?
because `SDL_Color *colors;` is a pointer to `SDL_Color`

https://metacpan.org/pod/FFI::Platypus::Record#FFI::Platypus::Record

```
typedef struct SDL_Color
{
    Uint8 r;
    Uint8 g;
    Uint8 b;
    Uint8 a;
} SDL_Color;
#define SDL_Colour SDL_Color

typedef struct SDL_Palette
{
    int ncolors;
    SDL_Color *colors;
    Uint32 version;
    int refcount;
} SDL_Palette;
```

And my you please provide an example in documentatino how to convert example above into perl?

**Comment by KES777** [comment]
Interesting question

**Comment by plicease** [comment]
This could use some additional documentation, but also some thought about the best location as to where to put it.

The TL;DR (mostly because it doesn't exist yet) is that:

 1. FFI::Platypus::Record should be used when passing small, non array of structures, non-nested, non-union structures that are passed by-value.
 2. FFI::C should be used pretty much any other situation where pass-by-value is not a requirement.

The other use cases aren't supported either by libffi or platypus:

 3. unions pass-by-value aren't supported by libffi
 4. nested structs pass-by-value probably won't ever be supported by Platypus.  The work around is to write a C wrapper that takes a pointer and passes that into the C code by-value (see `FFI::Platypus::Bundle`).

**Comment by plicease** [comment]
Hopefully this is more clearly documented in FFI::Platypus core, ::Record in FFI::C
